templates/commands/goldentest/record_test.go

```diff
@@ -303,6 +303,148 @@ kind: 'GoldenTest'`,
+				"template1/b.txt":                           "template 1 file B content",
+				"template1/testdata/golden/test1/test.yaml": testYaml,
+				"template2/spec.yaml":                       specYaml,
+				"template2/a.txt":                           "template 2 file A content",
+				"template2/b.txt":                           "template 2 file B content",
+				"template2/testdata/golden/test/test.yaml":  testYaml,
+			},
+			testNames: []string{"test1"},
+			expectedFileContent: map[string]string{
+				"template1/spec.yaml":                                specYaml,
+				"template1/a.txt":                                    "template 1 file A content",
+				"template1/b.txt":                                    "template 1 file B content",
+				"template1/testdata/golden/test1/test.yaml":          testYaml,
+				"template1/testdata/golden/test1/data/.abc/.gitkeep": "",
+				"template1/testdata/golden/test1/data/a.txt":         "template 1 file A content",
+				"template1/testdata/golden/test1/data/b.txt":         "template 1 file B content",
+				"template2/spec.yaml":                                specYaml,
+				"template2/a.txt":                                    "template 2 file A content",
+				"template2/b.txt":                                    "template 2 file B content",
+				"template2/testdata/golden/test/test.yaml":           testYaml,
+			},
+		},
+		{
+			name: "template1_record_err_template2_success",
+			filesContent: map[string]string{
+				"template1/spec.yaml":                      specYaml,
+				"template1/a.txt":                          "template 1 file A content",
+				"template1/b.txt":                          "template 1 file B content",
+				"template1/testdata/golden/test/test.yaml": "broken yaml",
+				"template2/spec.yaml":                      specYaml,
+				"template2/a.txt":                          "template 2 file A content",
+				"template2/b.txt":                          "template 2 file B content",
+				"template2/testdata/golden/test/test.yaml": testYaml,
+			},
+			expectedFileContent: map[string]string{
+				"template1/spec.yaml":                               specYaml,
+				"template1/a.txt":                                   "template 1 file A content",
+				"template1/b.txt":                                   "template 1 file B content",
+				"template1/testdata/golden/test/test.yaml":          "broken yaml",
+				"template2/spec.yaml":                               specYaml,
+				"template2/a.txt":                                   "template 2 file A content",
+				"template2/b.txt":                                   "template 2 file B content",
+				"template2/testdata/golden/test/test.yaml":          testYaml,
+				"template2/testdata/golden/test/data/.abc/.gitkeep": "",
+				"template2/testdata/golden/test/data/a.txt":         "template 2 file A content",
+				"template2/testdata/golden/test/data/b.txt":         "template 2 file B content",
+			},
+			wantErr: "failed to parse golden test",
+		},
```

Comment:
Could we add a test case for a template that has an output file named spec.yaml in its testdata?